nemo/export/tensorrt_llm.py

```diff
@@ -0,0 +1,707 @@
+            delete_existing_files (bool): if Truen, deletes all the files in model_dir.
+            n_gpus (int): number of GPUs to use for inference.
+            tensor_parallel_size (int): tensor parallelism.
+            pipeline_parallel_size (int): pipeline parallelism.
+            max_input_token (int): max input length.
+            max_output_token (int): max output length.
+            max_batch_size (int): max batch size.
+            max_prompt_embedding_table_size (int): max prompt embedding size.
+            use_inflight_batching (bool): if True, enables inflight batching for TensorRT-LLM Triton backend.
+            enable_context_fmha (bool): if True, use fused Context MultiHeadedAttention.
+            paged_kv_cache (bool): if True, uses kv cache feature of the TensorRT-LLM.
+            dtype (str): Floating point type for model weights (Supports BFloat16/Float16).
+            load_model (bool): load TensorRT-LLM model after the export.
+            enable_multi_block_mode (bool): enable faster decoding in multihead attention. Required for long context.
+        """
+
+        if model_type not in self.get_supported_models_list:
+            raise Exception(
+                "Model {0} is not currently a supported model type. "
+                "Supported model types are llama, gptnext, falcon, and starcoder".format(model_type)
+            )
+
+        if model_type == "gpt" or model_type == "starcoder":
+            model_type = "gptnext"
+
+        if model_type == "mixtral":
+            model_type = "llama"
+
+        if pipeline_parallel_size is None:
+            tensor_parallel_size = n_gpus
+            pipeline_parallel_size = 1
+        elif tensor_parallel_size is None:
+            tensor_parallel_size = 1
+            pipeline_parallel_size = n_gpus
+
+        if Path(self.model_dir).exists():
+            if delete_existing_files and len(os.listdir(self.model_dir)) > 0:
+                for files in os.listdir(self.model_dir):
+                    path = os.path.join(self.model_dir, files)
+                    try:
+                        shutil.rmtree(path)
+                    except OSError:
+                        os.remove(path)
+
+                if len(os.listdir(self.model_dir)) > 0:
+                    raise Exception("Couldn't delete all files.")
+            elif len(os.listdir(self.model_dir)) > 0:
+                raise Exception("There are files in this folder. Try setting delete_existing_files=True.")
+        else:
+            Path(self.model_dir).mkdir(parents=True, exist_ok=True)
+
+        if max_prompt_embedding_table_size is None:
+            max_prompt_embedding_table_size = 0
+
+        self.model = None
+
+        tmp_dir = tempfile.TemporaryDirectory()
+        nemo_export_dir = Path(tmp_dir.name)
+
+        model_configs, self.tokenizer = nemo_to_model_config(
```

Comment:
Please dont use generic names that equate `NeMo` to `NeMo LLM` - be explicit in function naming- `nemo_llm_to_model_config`

The term `NeMo` in this repo denotes all domains under NeMo as well as its core - ASR, CV, LLM, TTS, Multimoda - everything.

Reply:
Done.